### PR TITLE
test: Added Epic 3 Happy Path e2e test and refactoring

### PR DESCRIPTION
**Pull request by adamsheldon1**
## Description
* Added new e2e automated test to validate that 'add' happy path is working when uploading valid CSV and verifying records in BS_COHORT_DISTRIBUTION table.
* Refactored both tests to use Dependency Injection (DI) for improved maintainability and testability.
  - Injected 'AppSettings' for centralised config management across tests.
  - Injected helpers such as 'BlobstorageHelper' for reusability.
* Moved BlobStorageHelper to it's own dedicated class.
* Added 'ParquetHelperService' in preparation for transition from CSV to Parquet.
* Updated BaseIntegrationTest class, centralised critical config assertions.
* Added TestFiles sub-directories for 'add', 'update' and 'remove' for better organisation and added csv and parquet files to .gitignore to ensure they are not committed to the repository.
* General tidy up and improvements

## Context
We needed to add an E2E automated test to validate that a critical happy path flow for Epic 3 was working correctly.

## Type of changes

- [ ] Refactoring (non-breaking change)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [x] I have added tests to cover my changes
- [x] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: tests/IntegrationTests/Helpers/BaseIntegrationTest.cs [hunk 1]
```diff
@@ -1,39 +1,93 @@
...
         {
-            // Override this in derived classes to add additional setup if needed
-            return Task.CompletedTask;
+            bool isValid = true; // Overall flag to indicate if all critical configurations are valid
+
+            // Log the status of each configuration and service
+            if (AppSettings == null)
+            {
+                Logger.LogError("AppSettings configuration is not set.");
+                isValid = false;
```
Comment (MWClayson-NHS) [inline tests/IntegrationTests/Helpers/BaseIntegrationTest.cs:60]: you should probably just exit this method here and return a value that indicates failure.

File: tests/IntegrationTests/.gitignore [hunk 2]
```diff
@@ -0,0 +1,19 @@
+# Ignore appsettings.json
+appsettings.json
+
+# Ignore all CSV files in the IntegrationTests folder and its subdirectories
+*.csv
+
+# Ignore all Parquet files in the IntegrationTests folder and its subdirectories
+*.parquet
+
+# Ignore CSV and Parquet files in the TestFiles directory and its subdirectories
```
[the remark this answers is not in the document]
Comment (adamsheldon1) [reply tests/IntegrationTests/.gitignore:10]: Very true, my bad!

File: tests/IntegrationTests/Helpers/BaseIntegrationTest.cs (hunk 1 above)
Comment (adamsheldon1) [reply tests/IntegrationTests/Helpers/BaseIntegrationTest.cs:60]: Agreed, will refactor with early return pattern so it logs the first failure encountered for better control flow etc.

**Review by MWClayson-NHS — APPROVED** [review]
Looks good! Please can you hold off on merging until @SamRobinson75684 Has confirmed the azure test env is working.

**Pull request merged**